terminal.md

```diff
@@ -0,0 +1,505 @@
+
+# Unix/Linux Terminal
+
+ **Table of Contents**
+
+* [Opening the Terminal](#open-terminal)
+* [Keyboard Shortcuts](#keyboard-shortcuts)
+* [File System](#file-system)
+	* [Absolute and Relative paths](#absolute-and-relative-paths)
+* [Basics of Linux/Unix Commands](#command-basics)
+	* [Command Structure](#command-structure)
+	* [Command Help](#command-help)
+* [List of Commands](#command-list)
+
+## <a name="open-terminal"></a> Opening the Terminal
+
+  
+
+- Press <kbd>Win</kbd>
+
+- Type in `terminal` and press enter
+
+  
+
+OR
+
+ 
+
+- Press <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>T</kbd> (For Ubuntu and Linux Mint).
+
+### Windows users
+The WSL(Windows Subsystem for Linux) is a good alternative that lets developers run a GNU/Linux environment -- including most command-line tools, utilities, and applications -- directly on Windows, unmodified, without the overhead of a traditional virtual machine or dualboot setup. 
+<br>
+[Learn More](https://docs.microsoft.com/en-us/windows/wsl/) | [Install WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+  
+  
+
+## <a name="keyboard-shortcuts"></a> Keyboard shortcuts
+
+  
+
+- Use <b><kbd>Tab</kbd></b> to complete file & directory names and to further iterate over files. This can be done after partially typing out the name as well. 
+- Use <b><kbd>Shift</kbd> + <kbd>Tab</kbd></b> to go back to the previous names.
+- <b><kbd>&#8593;</kbd></b> or <b><kbd>Ctrl</kbd> + <kbd>P</kbd></b> to go to all the commands previously executed in that instance of the cmd, one by one. 
+- Similarly, <b><kbd>&#8595;</kbd></b> or <b><kbd>Ctrl</kbd> + <kbd>N</kbd></b>can be used to go towards the recent commands.
+- <b><kbd>Ctrl</kbd> + <kbd>R</kbd></b>
+- <b><kbd>Ctrl</kbd> + <kbd>&#8592;</kbd>/<kbd>&#8594;</kbd></b> to jump words.
+- <b><kbd>Home</kbd></b> or <b><kbd>Ctrl</kbd> + <kbd>A</kbd></b> to jump to the beginning and end of the line.
+- <b><kbd>End</kbd></b> or <b><kbd>Ctrl</kbd> + <kbd>E</kbd></b> keys to jump to the end of the line.
+-  <b><kbd>Ctrl</kbd> + <kbd>C</kbd></b> will stop running processes (like Node.js servers, etc) in your terminal and bring back the prompt (the cwd text on the command line).
+
```

Comment:
I think you should add a section on foreground and background processes, maybe as a separate PR:
- <kbd>Ctrl</kbd> + <kbd>Z</kbd> will pause the current running program.
- `fg` will resume it in the foreground.
- `bg` will let the program run in the background.
- <kbd>Ctrl</kbd> + <kbd>S</kbd> will stop command output.
- <kbd>Ctrl</kbd> + <kbd>Q</kbd> will resume command output.

Reply:
I thought of keeping the content as basic as possible so it's easier for beginners to follow. However, if you feel it's appropriate then it could be added as a separate PR.

Reply:
These commands are not too complicated, but are very, very useful when you have to work with remote systems and have only a single ssh connection.
I think you should add them.

Reply:
I see. So it'll be a new section at the end just like chaining commands?

Reply:
I would prefer a section right after `Keyboard Shortcuts`, called something like `Managing Processes`.

Reply:
Fine, so I'll add these in a separate PR?

Reply:
Yes.

Reply:
Cool, will do 👍